### PR TITLE
Added ROC curve util function

### PR DESCRIPTION
**Pull request by y27choi**
### Linked issue(s):
Fixes KOL-2649

### What change does this PR introduce and why?
* Added ROC curve util function for classification examples
* Copies `roc_curve` from `kolena.classification.multiclass`

### Please check if the PR fulfills these requirements

- [ ] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Relevant tests for the changes have been added
- [ ] Relevant docs have been added / updated

### CODE REVIEW

File: tests/unit/_experimental/classification/test_utils.py [hunk 1]
```diff
@@ -290,3 +290,152 @@ def test__compute_confusion_matrix__with_title() -> None:
...
+
+    # no positive samples
+    y_true = [0, 0, 0]
+    y_score = [0.1, 0.5, 0.6]
+    fpr, tpr = _roc_curve(y_true, y_score)
+    assert np.allclose(fpr, [0.0, 1 / 3, 1.0])
+    assert np.allclose(tpr, [])
+
+
+def test__compute_roc_curves__invalid() -> None:
```
Comment (ly-kolena) [inline tests/unit/_experimental/classification/test_utils.py]: Nit: what's the reason `pytest.mark.parametrize` is not used for this test?

Comment (y27choi) [reply tests/unit/_experimental/classification/test_utils.py]: hmm no particular reason.  It started off as a small test and more cases got added later.  Will use `parametrize`.

File: kolena/_experimental/classification/utils.py [hunk 2]
```diff
@@ -125,3 +154,98 @@ def compute_confusion_matrix(
...
+    labels: Optional[List[str]] = None,
+    title: Optional[str] = None,
+) -> Optional[CurvePlot]:
+    """
+    Computes OvR (one-vs-rest) ROC (receiver operating characteristic) curves for each class appears in `ground_truths`
+    if not specified.
+
+    :param ground_truths: The ground truth classification labels. For binary classification, the negative class is
+        `None`.
+    :param inferences: The inferences. For `N`-class problems, `label` is expected to contain `N` entries, one for
```
Comment (ly-kolena) [inline kolena/_experimental/classification/utils.py]: Nit: is it correct to use `label` in this explanation?

File: kolena/_experimental/classification/utils.py [hunk 3]
```diff
@@ -125,3 +154,98 @@ def compute_confusion_matrix(
...
+    if labels is None:
+        labels = sorted({gt.label for gt in ground_truths if gt is not None})
+
+    curves: List[Curve] = []
+    for label in labels:
+        y_true = [1 if gt is not None and gt.label == label else 0 for gt in ground_truths]
+        y_score = [get_label_confidence(label, inf) for inf in inferences]
+        fpr_values, tpr_values = _roc_curve(y_true=y_true, y_score=y_score)
+
+        if len(fpr_values) > 0 and len(tpr_values) > 0 and len(fpr_values) == len(tpr_values):
```
Comment (ly-kolena) [inline kolena/_experimental/classification/utils.py:237]: To my understanding, this identifies valid plot data. Is it useful to give users feedback for invalid path, or is `invalid` path a normal condition?

Comment (y27choi) [reply kolena/_experimental/classification/utils.py:237]: Yeah here the invalid path is when there is no negative sample or no positive sample in a test case.  IMO it's pretty easy to spot such cases by looking at the per class metric (`n_images`) which is why I decided not to log anything.

**Review by ly-kolena — APPROVED** [review]

**Pull request merged**